ext/QuantumOpticsExt/QuantumOpticsExt.jl

```diff
@@ -70,28 +71,67 @@ express_nolookup(s::XBasisState, ::QuantumOpticsRepr) = (_s₊,_s₋)[s.idx]
 express_nolookup(s::YBasisState, ::QuantumOpticsRepr) = (_i₊,_i₋)[s.idx]
 express_nolookup(s::ZBasisState, ::QuantumOpticsRepr) = (_l0,_l1)[s.idx]
 
-function express_nolookup(o::FockBasisState, r::QuantumOpticsRepr)
-    @warn "Fock space cutoff is not specified so we default to 2"
-    @assert o.idx<2 "without a specified cutoff you can not create states higher than 1 photon"
-    return (_f0₂,_f1₂)[o.idx+1]
+function express_nolookup(s::FockState, r::QuantumOpticsRepr)
+    b = basis(s)
+    i = s.idx
+    if !samebases(b, inf_fock_basis)
+        fockstate(b, i)
+    else
+        @warn "Fock space cutoff is not specified so we default to 2"
+        return fockstate(_bf2, i)
+    end
+end
```

Comment:
Let's **not** use the cutoff in `basis(symbolic_fock_expression)`. Symbolic expressions should always have an infinite cutoff, otherwise we have to make difficult decisions about what bases are equal or not, how to perform simplifications, etc.

Rather, we should add a field to `QuantumOpticsRepr`, e.g. `fock_cutoff::Union{Nothing,2}` that is used here. The decided cutoff should not be a property of the symbolic expression, rather a property of how the conversion is happening.

Reply:
The problem with raising a warning in the constructor is that it creates allocations and increases the runtime. This causes tests in `test_express_opt.jl` to not pass. I instead added a warning to the docs page; that should be enough, right?